docs/docs/learn/shards/working-with-data.md

```diff
@@ -1,11 +1,236 @@
+    (run main)
+    ```
+
+    1. `>=` is the alias for the [`Set`](../../../reference/shards/General/Set/) shard, with the parameter `Global` set to **false**. This makes `.x` a local variable.
+
+=== "Output"
+    ```{.clojure .annotate linenums="1"}
+    [get-x] x: 0
+    ```
+
+### Flow Methods
+
+If Wire Y is run from a separate Wire X using methods such as [`Do`](../../../reference/shards/General/Do), the variables on Wire X will be copied such that Y has access to its value at the moment it was called.
+
+However, this does not mean that Wire Y is in the same scope as X. Wire Y holds only a copy of the value - it does not have access to the actual variable.
+
+If a method such as [`Step`](../../../reference/shards/General/Step) is used instead, Wire Y would be scheduled on Wire X itself, giving it the same scope and access to X's actual variables.
+
+![Methods to control the Flow of Shards can affect scope.](assets/wires-scope.png)
+
+### Pure Wires
+
+Pure Wires are Wires that exist in their own scope. When run from another Wire, they do not copy that Wire's variables. 
+
+To create a Pure Wire, we use [`defpure`](../../../reference/functions/macros/#defpure).
+
+=== "Syntax"
+    ```{.clojure .annotate linenums="1"}
+    (defpure wire-name
+      ;; your shards here
+      )
+    ```
+
+In the example below, you can see how using `Step` on a Pure Wire still does not give it access to the parent Wire's variables.
+
+=== "Command"
+    ```{.clojure .annotate linenums="1"}
+    (defmesh main)
+
+    (defwire unpure-wire
+      (Get .x :Default 0) (Log))
+
+    (defpure pure-wire
+      (Get .x :Default 0) (Log))
+
+    (defwire main-wire
+      5 >= .x
+      (Step unpure-wire)
+      (Step pure-wire))
+
+    (schedule main main-wire)
+    (run main)
+    ```
+
+=== "Output"
+    ```
+    [unpure-wire] 5
+    [pure-wire] 0
+    ```
+
+### Defined Constants
+
+If you define a constant in your program, it will have a global scope and can be accessed by any Wire in your program.
+
+=== "Command"
+    ```{.clojure .annotate linenums="1"}
+    (defmesh main)
+
+    (def x 1)
+
+    (defwire get-x
+      x (Log "x"))
+
+    (schedule main get-x)
+    (run main)
+    ```
+
+=== "Output"
+    ```{.clojure .annotate linenums="1"}
+    [get-x] x: 1
+    ```
+
+!!! note
+    Note that the constant `x` defined in the example above is named differently from variables, which would have been `.x` in this case.
+
+## Passthrough
+
+Passthrough determines if data can pass through a Wire unaltered. It allows you to better control the state of the data moving through your program.
+
+Wires will have passthrough disabled by default. In order to enable passthrough for Wires, we employ the shard [`Sub`](../../../reference/shards/General/Sub/). Any shards passed into the `Shards` parameter of `Sub` will run as per usual, except that the final output will be replaced with the initial input passed into `Sub`, thereby creating a passthrough effect.
+
```

Comment:
It is not really the wire that has "passthrough" disabled, but more that most shards consume the data and produce another value. So the only way to "reuse" the same value multiple times is to either save it to a variable or to use `Sub` (or `|`). I don't think we should talk about "wire" in this context. Letting @sinkingsugar and @guusw give their opinion.

On that "reuse" note, I would like to have another equivalent sample to show that `Sub` allows to "save" a temporary variable.

```
=== "Example with temporary variable"
    ```{.clojure .annotate linenums="1"}
    (defmesh main)

    (defwire temp-test
      1 (Log "Before")
      >= .temp
      (Math.Add 2) >= .x
      .x (Log "In")
      .temp
      (Log "After"))
    (schedule main  temp-test)
    (run main)
    ```
```